File: problems/0973-k-closest-points-to-origin/typescript/solution.ts
// K Closest Points to Origin — https://leetcode.com/problems/k-closest-points-to-origin/
export function kClosest(points: number[][], k: number): number[][] {
  const dist = (p: number[]): number => p[0] * p[0] + p[1] * p[1];
  return [...points].sort((a, b) => dist(a) - dist(b)).slice(0, k);
}

File: problems/0973-k-closest-points-to-origin/go/solution.go
package solution

import "sort"

// K Closest Points to Origin — https://leetcode.com/problems/k-closest-points-to-origin/
func kClosest(points [][]int, k int) [][]int {
	sort.Slice(points, func(i, j int) bool {
		di := points[i][0]*points[i][0] + points[i][1]*points[i][1]
		dj := points[j][0]*points[j][0] + points[j][1]*points[j][1]
		return di < dj
	})
	return points[:k]
}
